src/compile-middleware.ts

```diff
@@ -141,15 +141,19 @@ const isInlineJavaScript = dom5.predicates.AND(
     dom5.predicates.hasTagName('script'),
     dom5.predicates.NOT(dom5.predicates.hasAttr('src')));
 
-function needCompilation(uaParser: UAParser): boolean {
-  const browser = uaParser.getBrowser();
+export function needCompilation(userAgent: string): boolean {
+  const browser = new UAParser(userAgent).getBrowser();
   const versionSplit = browser.version && browser.version.split('.');
   const majorVersion = versionSplit ? parseInt(versionSplit[0], 10) : -1;
+  const minorVersion = versionSplit && versionSplit.length > 1 ?
```

Comment:
**ES6 golf** this could be:
```javascript
const [majorVersion, minorVersion] = versionSplit.map((v) => parseInt(v, 10));
```

But we need to drop node v4 support, which we've started in other repos